File: app/src/main/java/com/alexsheiko/invitationmaker/MainActivity.java
package com.alexsheiko.invitationmaker;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.alexsheiko.invitationmaker.base.BaseActivity;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @SuppressWarnings("unused")
    public void onClickCategory(View view) {
        Intent intent = new Intent(this, GridActivity.class);
        for (int i = 0; i < ((ViewGroup) view).getChildCount(); ++i) {
            View child = ((ViewGroup) view).getChildAt(i);
            if (child instanceof TextView) {
                String category = ((TextView) child).getText().toString()
                        .replaceAll(" ", "").replace("\n", " ");
                intent.putExtra("category", category);
            }
        }
        startActivity(intent);
    }
}

File: app/src/main/java/com/alexsheiko/invitationmaker/MainActivity.kt
package com.alexsheiko.invitationmaker

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.alexsheiko.invitationmaker.base.BaseActivity
import org.jetbrains.anko.alert

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClickCategory(view: View) {
        val intent = Intent(this, GridActivity::class.java)
        (0..(view as ViewGroup).childCount - 1)
                .map { view.getChildAt(it) }
                .filterIsInstance<TextView>()
                .map {
                    it.text.toString()
                            .replace(" ", "").replace("\n", " ")
                }
                .forEach { intent.putExtra("category", it) }
        startActivity(intent)
    }

    override fun onBackPressed() {
        alert("Are you sure you want to exit the app?", "Stay with us") {
            positiveButton("Exit") { super.onBackPressed() }
            negativeButton("Stay on") { }
        }.show()
    }
}
